src/core/io/src/4C_io_mesh.hpp

```diff
@@ -517,6 +524,45 @@ namespace Core::IO::MeshInput
           data);
     }
   }
+
+  /*!
+   * @brief Reads in a map with value_type @p T from the point data of the given @p mesh.
+   *
+   * @throws if the point data is not found in all point sets or if the type of the point data is
+   * not implicitly convertible to the expected type.
+   */
+  template <typename IndexType, typename T, unsigned dim>
+    requires(std::is_integral_v<IndexType>)
+  void read_value_from_point_data(
+      const Mesh<dim>& mesh, const std::string& key, std::unordered_map<IndexType, T>& value)
+  {
+    for (const auto& point_ref : mesh.points_with_data())
+    {
+      FOUR_C_ASSERT_ALWAYS(point_ref.contains_data(key),
+          "The mesh does not contain point data with the name '{}'.", key);
```

Comment:
Related to the other comment, this check only needs to happen once outside the loop and on the mesh.